tests/multinode.rs

```diff
@@ -67,30 +39,49 @@ fn converge(
         spy_window,
         spy.sockets.gossip,
         spy.sockets.gossip_send,
-        exit,
+        exit.clone(),
     ).unwrap();
     //wait for the network to converge
     let mut converged = false;
+    let mut rv = vec![];
     for _ in 0..30 {
         let num = spy_ref.read().unwrap().convergence();
-        if num == num_nodes as u64 {
+        let mut v: Vec<ReplicatedData> = spy_ref
+            .read()
+            .unwrap()
+            .table
+            .values()
+            .into_iter()
+            .filter(|x| x.id != me)
+            .filter(|x| x.requests_addr != daddr)
```

Comment:
You'll want to merge these yourself. This isn't Haskell, sorry.

Reply:
i think it will actually fuse.  the filter returns a iterator trait, so the compiled code should be a bunch of static functions calling each other which should get inlined

Reply:
yea, check this out
```
 INFO 2018-07-03T18:55:36Z: solana::fullnode: processing ledger...                                                                                            
 INFO 2018-07-03T18:55:36Z: solana::fullnode: ENTRY                                                                                                           
 INFO 2018-07-03T18:55:36Z: solana::fullnode: ENTRY  
```
from
```
                let entries = entry_writer::read_entries(&mut r).map(|e| {
                    info!("ENTRY");
                    e.expect("failed to parse entry")
                });
                info!("processing ledger...");
                bank.process_ledger(entries).expect("process_ledger")
```
At least the execution is definitely delayed until iteration.